kube-runtime/src/reflector/object_ref.rs

```diff
@@ -141,10 +140,12 @@ pub struct ObjectRef<K: Lookup + ?Sized> {
     ///
     /// This is *not* considered when comparing objects, but may be used when converting to and from other representations,
     /// such as [`OwnerReference`] or [`ObjectReference`].
-    #[derivative(Hash = "ignore", PartialEq = "ignore")]
+    #[educe(Hash(ignore), PartialEq(ignore))]
     pub extra: Extra,
 }
 
+impl<K: Lookup + ?Sized> Eq for ObjectRef<K> where K::DynamicType: Eq {}
```

Comment:
The `Eq` relation is actually the only thing I was a little worried about here because it is [semi-implicitly](https://github.com/kube-rs/kube/blob/b57cf5d2ef3ce954063fc4260ab051a251f1b6c2/kube-runtime/src/reflector/store.rs#L12C37-L12C46) relied on to prevent duplicates in the store in the store, but thankfully [this relation is tested](https://github.com/kube-rs/kube/blob/b57cf5d2ef3ce954063fc4260ab051a251f1b6c2/kube-runtime/src/reflector/object_ref.rs#L312-L363).

Reply:
Good to know this is tested, I also wasn't too sure. This line is basically copied from the expansion of the former 'derivative' macro. For reference, this is what derivative generates:

```rs
#[allow(unused_qualifications)]
impl<K: Lookup + ?Sized> ::std::cmp::Eq for ObjectRef<K> where K::DynamicType: Eq {}
```